drivers/dai/intel/ssp/ssp.c

```diff
@@ -18,24 +18,119 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
-#define dai_mn_base(dai) dai->plat_data.mn_inst->base
-#define dai_base(dai) dai->plat_data.base
-#define dai_ip_base(dai) dai->plat_data.ip_base
-#define dai_shim_base(dai) dai->plat_data.shim_base
-#define dai_hdamlssp_base(dai) dai->plat_data.hdamlssp_base
-#define dai_i2svss_base(dai) dai->plat_data.i2svss_base
+#define dai_get_plat_data(dai) dai->ssp_plat_data
+#define dai_get_mn(dai) dai->ssp_plat_data->mn_inst
+#define dai_get_ftable(dai) dai->ssp_plat_data->ftable
+#define dai_get_fsources(dai) dai->ssp_plat_data->fsources
+#define dai_mn_base(dai) dai->ssp_plat_data->mn_inst->base
+#define dai_base(dai) dai->ssp_plat_data->base
+#define dai_ip_base(dai) dai->ssp_plat_data->ip_base
+#define dai_shim_base(dai) dai->ssp_plat_data->shim_base
+#define dai_hdamlssp_base(dai) dai->ssp_plat_data->hdamlssp_base
+#define dai_i2svss_base(dai) dai->ssp_plat_data->i2svss_base
 
 #define DAI_DIR_PLAYBACK 0
 #define DAI_DIR_CAPTURE 1
 #define SSP_ARRAY_INDEX(dir) dir == DAI_DIR_RX ? DAI_DIR_CAPTURE : DAI_DIR_PLAYBACK
 
+static const char irq_name_level5_z[] = "level5";
+
+static struct dai_intel_ssp_freq_table ssp_freq_table[] = {
+	{ DT_PROP(DT_NODELABEL(audioclk), clock_frequency),
+	  DT_PROP(DT_NODELABEL(audioclk), clock_frequency) / 1000},
+	{ DT_PROP(DT_NODELABEL(sysclk), clock_frequency),
+	  DT_PROP(DT_NODELABEL(sysclk), clock_frequency) / 1000},
+	{ DT_PROP(DT_NODELABEL(pllclk), clock_frequency),
+	  DT_PROP(DT_NODELABEL(pllclk), clock_frequency) / 1000},
+};
+
+static uint32_t ssp_freq_sources[] = {
+	DAI_INTEL_SSP_CLOCK_AUDIO_CARDINAL,
+	DAI_INTEL_SSP_CLOCK_XTAL_OSCILLATOR,
+	DAI_INTEL_SSP_CLOCK_PLL_FIXED,
+};
+
+static struct dai_intel_ssp_mn ssp_mn_divider = {
+	.base = DT_REG_ADDR_BY_IDX(DT_NODELABEL(ssp0), 1),
+};
+
+
+#define INTEL_SSP_INST_DEFINE(node_id) {			\
+		.is_initialized = false, \
+		.is_power_en = false, \
+		.acquire_count = 0, \
+		.ssp_index = DT_PROP(node_id, ssp_index), \
+		.base =	DT_REG_ADDR_BY_IDX(node_id, 0),			\
+		IF_ENABLED(DT_NODE_EXISTS(DT_NODELABEL(sspbase)),				\
+		(.ip_base = DT_REG_ADDR_BY_IDX(DT_NODELABEL(sspbase), 0),))		\
+		.shim_base = DT_REG_ADDR_BY_IDX(DT_NODELABEL(shim), 0),			\
+		IF_ENABLED(DT_NODE_EXISTS(DT_NODELABEL(hdamlssp)),				\
+			(.hdamlssp_base = DT_REG_ADDR(DT_NODELABEL(hdamlssp)),))	\
+		IF_ENABLED(DT_PROP_HAS_IDX(node_id, i2svss, 0),					\
+			(.i2svss_base = DT_PROP_BY_IDX(node_id, i2svss, 0),))		\
+		.irq = DT_NUM_IRQS(node_id), \
+		.irq_name = irq_name_level5_z,					\
+		.fifo[DAI_DIR_PLAYBACK].offset =				\
+			DT_REG_ADDR_BY_IDX(node_id, 0) + SSDR,		\
+		.fifo[DAI_DIR_PLAYBACK].handshake =				\
+			DT_DMAS_CELL_BY_NAME(node_id, tx, channel),	\
+		.fifo[DAI_DIR_CAPTURE].offset =					\
+			DT_REG_ADDR_BY_IDX(node_id, 0) + SSDR,		\
+		.fifo[DAI_DIR_CAPTURE].handshake =				\
+			DT_DMAS_CELL_BY_NAME(node_id, rx, channel), \
+		.mn_inst = &ssp_mn_divider,			\
+		.ftable = ssp_freq_table,			\
+		.fsources = ssp_freq_sources,		\
+		.clk_active = 0,					\
+},
```

Comment:
Your editor likely has the tab size set to 4 spaces. If possible, please change it to 8 and correct it.

Reply:
fixed